src/osp/Active/SysRender.cpp

```diff
@@ -0,0 +1,275 @@
+    fbo.attachRenderbuffer(GL::Framebuffer::BufferAttachment::DepthStencil, *depthStencilRes);
+
+    DependRes<GL::Framebuffer> fboRes =
+        resources.add<GL::Framebuffer>("offscreen_fbo", std::move(fbo));
+
+    rScene.reg_emplace<ACompRenderTarget>(rScene.hier_get_root(),
+        viewSize, std::move(fboRes));
+    rScene.reg_emplace<ACompFBOColorAttachment>(rScene.hier_get_root(), colorRes);
+
+
+    // Create the default render pipeline
+    resources.add<RenderPipeline>("default",
+        std::move(create_render_pipeline()));
+}
+
+RenderPipeline SysRender::create_render_pipeline()
+{
+    /* Define render passes */
+    RenderOrder_t pipeline;
+    std::vector<RenderOrderHandle_t> handles;
+
+    // Opaque pass
+    handles.emplace_back(pipeline,
+        "opaque_pass", "", "transparent_pass",
+        [](ActiveScene& rScene, ACompCamera const& camera)
+        {
+            using namespace Magnum;
+            using Magnum::GL::Renderer;
+
+            auto& reg = rScene.get_registry();
+
+            Renderer::enable(Renderer::Feature::DepthTest);
+            Renderer::enable(Renderer::Feature::FaceCulling);
+            Renderer::disable(Renderer::Feature::Blending);
+
+            // Fetch opaque objects
+            auto opaqueView = reg.view<ACompOpaque, ACompVisible, ACompShader>();
+
+            SysRender::draw_group(rScene, opaqueView, camera);
+        }
+    );
+
+    // Transparent pass
+    handles.emplace_back(pipeline,
+        "transparent_pass", "opaque_pass", "display_framebuffer",
+        [](osp::active::ActiveScene& rScene, ACompCamera const& camera)
+        {
+            using Magnum::GL::Renderer;
+            using namespace osp::active;
+
+            auto& reg = rScene.get_registry();
+
+            Renderer::enable(Renderer::Feature::DepthTest);
+            Renderer::enable(Renderer::Feature::FaceCulling);
+            Renderer::enable(Renderer::Feature::Blending);
+            Renderer::setBlendFunction(
+                Renderer::BlendFunction::SourceAlpha,
+                Renderer::BlendFunction::OneMinusSourceAlpha);
+
+            auto transparentView = reg.view<ACompTransparent, ACompVisible, ACompShader>();
+
+            // Draw backfaces
+            Renderer::setFaceCullingMode(Renderer::PolygonFacing::Front);
+            SysRender::draw_group(rScene, transparentView, camera);
+
+            // Draw frontfaces
+            Renderer::setFaceCullingMode(Renderer::PolygonFacing::Back);
+            SysRender::draw_group(rScene, transparentView, camera);
+        }
+    );
+
+    return {pipeline};
+}
+
+void SysRender::update(ActiveScene& rScene)
+{
+    auto view = rScene.get_registry().view<ACompVisible>();
+    for (ActiveEnt e : view)
+    {
+        auto* shader = rScene.reg_try_get<ACompShader>(e);
+
+        if (shader == nullptr)
+        {
+            // Drawable entity has not been initialized yet
+            constexpr ShaderDrawFnc_t defaultDraw = shader::Phong::draw_entity;
```

Comment:
imho not really useful to have this defaultDraw variable.

Reply:
I separated it out like that because I'm not sure if I'm going to just hard-code it, or if it should be fetched from elsewhere.